src/Bicep.Core/TypeSystem/Az/AzResourceTypeLoaderFactory.cs

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.IO;
+using Bicep.Core.Registry.Oci;
+using Newtonsoft.Json;
+using Bicep.Core.Features;
+
+namespace Bicep.Core.TypeSystem.Az
+{
+    public class AzResourceTypeLoaderFactory : IAzResourceTypeLoaderFactory
+    {
+        private const string typesArtifactFilename = "types.tgz";
+
+        private readonly IFeatureProviderFactory featureProviderFactory;
+
+        private Dictionary<string, IAzResourceTypeLoader> resourceTypeLoaders;
+
+        public AzResourceTypeLoaderFactory(IFeatureProviderFactory featureProviderFactory, IAzResourceTypeLoader defaultAzResourceTypeLoader)
+        {
+            this.featureProviderFactory = featureProviderFactory;
+            this.resourceTypeLoaders = new() {
+                {"builtin", defaultAzResourceTypeLoader},
+            };
+        }
+
+        public IAzResourceTypeLoader GetBuiltInTypeLoader()
+        {
+            return resourceTypeLoaders["builtin"];
```

Comment:
Could you define "builtin" as a const in this file, so that finding references is obvious?

```csharp
private const string BuiltInLoaderKey = "builtin";
```